tests/test_base.py

```diff
@@ -121,3 +127,12 @@ def test_lines_iterator_noqa(hook_iterator, capsys):
     assert lines[0] == 'more text'
     assert lines[1] == ''
     assert result is True
+
+
+def test_lines_iterator_prints_filename_on_invalid_files(hook_iterator):
+    with mock.patch('hulks.base.open') as mocked_open:
+        mocked_open.return_value = UnicodeErrorRaiserIO(b'This is invalid utf-8:\xfe\xfe!')
```

Comment:
`mocked_open.side_effect = ...` would be more appropriate in this case

Reply:
No, it did not work-  I just spent a while trying to use `mock.side_effect` here, after your comment, and it turned out that it is not the "file opening" that have to raise the error, rather:

-  "open" in the target namespace have to return an object that can answer to the ContextManager protocol,
- its ` __enter__` method return an object that sports a `readlines` method 
- This readlines that when being iterated upon will raise the unicode error.

 Even moving the error to when `readlines` is called rather than iterated, I ended up trying to build a mock object that is far more complicated than the four-liner error raiser I created above.

Reply:
Have you tried `mock.MagicMock` ? It does implement `__enter__`, among others.

Reply:
The correct way of mocking this is: `mocked_open().__enter__().readlines.side_effect = TypeError`*. It's not pretty, but it does saves us some lines

Reply:
The question is: are all this magic of Mock objects really "better"? I already choke at "site_effect" being one special attribute name, among the infinite other generic attribute names.
Sometimes the "it is not pretty" is just too ugly. :-)

Reply:
Let ś just say  that "mock is too magic" sometimes.

I am really trying to use it - it is not getting more readable, and I am getting nauseated at mock bad semantics.

Now, here is what I am fighting against -
I have this at the test:

        mocked_open.__enter__().readlines.side_effect = UnicodeDecodeError('utf-8', b'', 0, 0, 'invalid start byte')

And on running, this is on the test frame:
```
(Pdb) mocked_open
<MagicMock name='open' id='140639207535616'>
(Pdb) mocked_open.__enter__()
<MagicMock name='open.__enter__()' id='140639207535952'>
(Pdb) mocked_open.__enter__().readlines
<MagicMock name='open.__enter__().readlines' id='140639207130280'>
(Pdb) mocked_open.__enter__().readlines.side_effect
UnicodeDecodeError('utf-8', b'', 0, 0, 'invalid start byte')
(Pdb) down
(...) # all the way down to the "lines_iterator" frame
> /home/gwidion/olist/hulks/hulks/base.py(38)lines_iterator()
-> for line_number, line in enumerate(lines, 1):
(Pdb) fp
<MagicMock name='open().__enter__()' id='140639207155080'>
(Pdb) fp.side_effect
(Pdb) fp.readlines.side_effect
(Pdb) 
```
As you can see the object returned by `__enter__` here is the one returned by `__enter__.side_effect` on the test frame - "magic mocking" taking place. And neither this object, nor the one returned at the `.readlines` have `side_effect` set. 

And moreover, the mock won't work with `iter` and the code would have to stick with `readlines`.

It is simply not worth it - mock is to be a helper tool, and in this case it is clearly getting in the way.